staging/src/k8s.io/component-base/config/types.go

```diff
@@ -39,6 +39,9 @@ type ClientConnectionConfiguration struct {
 // LeaderElectionConfiguration defines the configuration of leader election
 // clients for components that can run with leader election enabled.
 type LeaderElectionConfiguration struct {
+	// clientTimeout indicates the timeout set for client used to communicate
+	// with kube-apiserver.
+	ClientTimeout metav1.Duration
```

Comment:
I think that overall this parameter makes sense to me.
However, I'm not convinced that it should really be defaulted to 2s (in defaulting).

In my personal opinion what we should do is:
- add the field here
- default it to somewhat what we were previously doing (so either to RenewDeadline, or maybe 10s which is its default value)
- start respecting it

And in the second PR, switch it to 2s (or whatever else value) in kcm, ccm and scheduler.

But given it's an API change, I would like to hear @liggitt `s thoughts on that too.